addons/godot-firebase/database/reference.gd

```diff
@@ -148,7 +148,7 @@ func get_data() -> Dictionary:
 
 func _get_remaining_path(is_push: bool = true) -> String:
     var remaining_path = ""
-    if not _filter_query or is_push:
+    if _filter_query == null or _filter_query.is_empty() or !is_push:
```

Comment:
Is there a reason you added the not on is_push here? That changes the logic. Previously, the not was only on _filter_query, meaning it was checking to see if the filter query was not null. This confuses me though - does this fix a bug that was there? I think we still don't want the not on the is_push, but the rest of the changes seem like they must be right, as the logic otherwise would require the filter, which doesn't make sense if the filter is null (which is what would be the case in the original). So remove the not from is_push and this should work, I think.